meshmode/mesh/processing.py

```diff
@@ -1538,4 +1596,36 @@ def make_mesh_grid(
 
 # }}}
 
+
+# {{{ remove_unused_vertices
+
+def remove_unused_vertices(mesh: Mesh) -> Mesh:
+    if mesh.vertices is None:
+        raise ValueError("mesh must have vertices")
+
+    def not_none(vi: Optional[np.ndarray]) -> np.ndarray:
+        if vi is None:
+            raise ValueError("mesh element groups must have vertex indices")
+        return vi
+
+    used_vertices = np.unique(np.sort(np.concatenate([
+        not_none(grp.vertex_indices).reshape(-1)
+        for grp in mesh.groups
+    ])))
+
+    used_flags: np.ndarray = np.zeros(mesh.nvertices, dtype=np.bool_)
+    used_flags[used_vertices] = 1
+    new_vertex_indices = np.cumsum(used_flags.astype(mesh.vertex_id_dtype)) - 1
```

Comment:
```suggestion
    new_vertex_indices = np.cumsum(used_flags, dtype=mesh.vertex_id_dtype) - 1
```
? I think that's the same, not sure if it makes any difference..

Reply:
It actually isn't the same; it seems that's the only way to influence the `cumsum` output dtype.